cudax/include/cuda/experimental/__container/vector.cuh

```diff
@@ -0,0 +1,1585 @@
+
+  //! @brief Changes the size of the \c vector to \p __size and leaves new elements uninitialized
+  //! @param __size The intended size of the vector.
+  //! If `__size < vec.size()` then it destroys all superfluous elements. Otherwise, it provides sufficient storage and
+  //! adjusts the size of the vector, but leaves the new elements uninitialized.
+  void resize(const size_type __size, uninit_t) noexcept
+  {
+    if (__size <= __size_)
+    {
+      // nothing to do
+    }
+    else
+    {
+      if (capacity() < __size)
+      {
+        __buffer __new_buf{get_resource(), __size};
+        this->__copy_same(__unwrapped_begin(), __unwrapped_end(), __new_buf.begin());
+        __buf_.__swap_allocations(__new_buf);
+      }
+    }
+    __size_ = __size;
+  }
+
+  //! @brief Reallocates the storage, so that `vec.capacity() == vec.size()`
+  //! If `vec.size() == vec.capacity()` this is a noop. If `vec.empty()` holds, then no storage is allocated
+  void shrink_to_fit() noexcept
+  {
+    if (__size_ == capacity())
+    {
+      return;
+    }
+
+    __buffer __new_buf{get_resource(), __size_};
+    this->__copy_same(__unwrapped_begin(), __unwrapped_end(), __new_buf.begin());
+    __buf_.__swap_allocations(__new_buf);
+  }
+
+  //! @brief Swaps the contents of a vector with those of \p __other
+  //! @param __other The other vector.
+  void swap(vector& __other) noexcept
+  {
+    __buf_.swap(__other.__buf_);
+    _CUDA_VSTD::swap(__size_, __other.__size_);
+  }
+
+  //! @brief Swaps the contents of two vectors
+  //! @param __lhs One vector.
+  //! @param __rhs The other vector.
+  friend void swap(vector& __lhs, vector& __rhs) noexcept
+  {
+    __lhs.swap(__rhs);
+  }
+  //! @}
+
+  //! @addtogroup comparison
+  //! @{
+
+  //! @brief Compares two vectors for equality
+  //! @param __lhs One vector.
+  //! @param __rhs The other vector.
+  //! @return true, if \p __lhs and \p __rhs contain equal elements have the same size
+  _CCCL_NODISCARD_FRIEND constexpr bool
+  operator==(const vector& __lhs, const vector& __rhs) noexcept(noexcept(_CUDA_VSTD::equal(
+    __lhs.__unwrapped_begin(), __lhs.__unwrapped_end(), __rhs.__unwrapped_begin(), __rhs.__unwrapped_end())))
+  {
+    return vector::__equality(
+      __lhs.__unwrapped_begin(), __lhs.__unwrapped_end(), __rhs.__unwrapped_begin(), __rhs.__unwrapped_end());
+  }
+#if _CCCL_STD_VER <= 2017
+  //! @brief Compares two vectors for inequality
+  //! @param __lhs One vector.
+  //! @param __rhs The other vector.
+  //! @return false, if \p __lhs and \p __rhs contain equal elements have the same size
+  _CCCL_NODISCARD_FRIEND constexpr bool
+  operator!=(const vector& __lhs, const vector& __rhs) noexcept(noexcept(_CUDA_VSTD::equal(
+    __lhs.__unwrapped_begin(), __lhs.__unwrapped_end(), __rhs.__unwrapped_begin(), __rhs.__unwrapped_end())))
+  {
+    return !vector::__equality(
+      __lhs.__unwrapped_begin(), __lhs.__unwrapped_end(), __rhs.__unwrapped_begin(), __rhs.__unwrapped_end());
+  }
```

Comment:
there's no need for these to be `constexpr`. there are no `constexpr` constructors.

Reply:
True but it does not hurt?